src/go/k8s/apis/redpanda/v1alpha1/cluster_types.go

```diff
@@ -129,7 +129,19 @@ type ClusterSpec struct {
 	// List of superusers
 	Superusers []Superuser `json:"superUsers,omitempty"`
 	// SASL enablement flag
+	// +optional
+	// Deprecated: replaced by "kafkaEnableAuthorization"
```

Comment:
This is unfortunately not possible to prevent from happening. The combination of `EnableSASL: true, KafkaEnableAuthorization:false` will be possible until core removes `enableSASL` configuration option.